cli/pkg/kctrl/cmd/package/installed/package_install_dry_run.go

```diff
@@ -0,0 +1,147 @@
+package installed
+
+import (
+	"fmt"
+
+	cmdcore "github.com/vmware-tanzu/carvel-kapp-controller/cli/pkg/kctrl/cmd/core"
+	kcpkgv1alpha1 "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apis/packaging/v1alpha1"
+	versions "github.com/vmware-tanzu/carvel-vendir/pkg/vendir/versions/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+)
+
+type PackageInstalledDryRun struct {
+	*CreateOrUpdateOptions
+}
+
+func (d PackageInstalledDryRun) PrintResources() error {
+	const yamlSeperator = "---\n"
+	packageInstall := &kcpkgv1alpha1.PackageInstall{
```

Comment:
We seem to be duplicating a lot of code from create_or_update, maybe those sections should be taken out as separate functions to be reused here as well?

Reply:
That was the original intention, however all the existing code relies on a client. The logic was becoming quite branched for one feature. (Conditionally check if dry run is being used every time we use a client and every time we print something).
The names are derived from the shared `createdAnnotations` resource and I have reused the function managing the annotations added to the pkgi.

Furthermore, since these APIs are stable I see the resource creation logic not evolving a lot over time 🤔